drivers/serial/uart_stm32.c

```diff
@@ -1387,6 +1390,65 @@ static int uart_stm32_init(const struct device *dev)
 #endif
 }
 
+#ifdef CONFIG_PM_DEVICE
```

Comment:
`uart_stm32_pm_control()` is now also part of this #ifdef. Could this cause problems when `CONFIG_PM_DEVICE` isn't selected?

Reply:
In case the PM_DEVICE is not defined, the uart_stm32_pm_control does not exist but  DEVICE_DT_INST_DEFINE get a null value for this entry (no error).